cmd/zoekt-dynamic-indexserver/main.go

```diff
@@ -193,6 +229,31 @@ func (s *indexServer) serveTruncate(w http.ResponseWriter, r *http.Request) {
 	s.incrementRequestsTotal(r.Method, route, http.StatusOK)
 }
 
+func (s *indexServer) serveDelete(w http.ResponseWriter, r *http.Request) {
+	route := "delete"
+	dec := json.NewDecoder(r.Body)
+	dec.DisallowUnknownFields()
+
+	var req deleteRequest
+	if err := dec.Decode(&req); err != nil {
+		log.Printf("Error decoding delete request: %v", err)
+		http.Error(w, "JSON parser error", http.StatusBadRequest)
+		return
+	}
+
+	err := deleteRepository(s.opts, req.RepoID)
```

Comment:
don't you also need to delete the shards for repo, or will you let converge by the indexing server noticing it is gone.